athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/GcsMetadataHandler.java

```diff
@@ -0,0 +1,290 @@
+import java.util.stream.Collectors;
+
+import static com.amazonaws.athena.connector.lambda.metadata.ListTablesRequest.UNLIMITED_PAGE_SIZE_VALUE;
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.CLASSIFICATION_GLUE_TABLE_PARAM;
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.GCS_SECRET_KEY_ENV_VAR;
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.STORAGE_SPLIT_JSON;
+import static java.util.Objects.requireNonNull;
+
+public class GcsMetadataHandler
+        extends GlueMetadataHandler
+{
+    public static final String TABLE_FILTER_IDENTIFIER = "gs://";
+    private static final Logger LOGGER = LoggerFactory.getLogger(GcsMetadataHandler.class);
+    /**
+     * used to aid in debugging. Athena will use this name in conjunction with your catalog id
+     * to correlate relevant query errors.
+     */
+    private static final String SOURCE_TYPE = "gcs";
+    //as this connector mandatory required glue database and table metadata
+    private static final boolean DISABLE_GLUE = false;
+    private static final CharSequence GCS_FLAG = "google-cloud-storage-flag";
+    private static final DatabaseFilter DB_FILTER = (Database database) -> (database.getLocationUri() != null && database.getLocationUri().contains(GCS_FLAG));
+    // used to filter out Glue tables which lack indications of being used for GCS.
+    private static final TableFilter TABLE_FILTER = (Table table) -> table.getStorageDescriptor().getLocation().startsWith(TABLE_FILTER_IDENTIFIER);
+    private final StorageMetadata datasource;
+    private final AWSGlue glueClient;
+    private BufferAllocator allocator;
+
+    public GcsMetadataHandler(BufferAllocator allocator) throws IOException
+    {
+        super(DISABLE_GLUE, SOURCE_TYPE);
+        String gcsCredentialsJsonString = this.getSecret(System.getenv(GCS_SECRET_KEY_ENV_VAR));
+        this.datasource = new StorageMetadata(gcsCredentialsJsonString);
+        this.glueClient = getAwsGlue();
+        requireNonNull(glueClient, "Glue Client is null");
+        this.allocator = allocator;
+    }
+
+    @VisibleForTesting
+    @SuppressWarnings("unused")
+    protected GcsMetadataHandler(EncryptionKeyFactory keyFactory,
+                                 AWSSecretsManager awsSecretsManager,
+                                 AmazonAthena athena,
+                                 String spillBucket,
+                                 String spillPrefix,
+                                 AmazonS3 amazonS3, AWSGlue glueClient, BufferAllocator allocator) throws IOException
+    {
+        super(glueClient, keyFactory, awsSecretsManager, athena, SOURCE_TYPE, spillBucket, spillPrefix);
+        String gcsCredentialsJsonString = this.getSecret(System.getenv(GCS_SECRET_KEY_ENV_VAR));
+        this.datasource = new StorageMetadata(gcsCredentialsJsonString);
+        this.glueClient = getAwsGlue();
+        requireNonNull(glueClient, "Glue Client is null");
+        this.allocator = allocator;
```

Comment:
this is copy pasted from the other constructor, let's try to avoid repeating blocks.

Reply:
Hi @hackett123 , I can not find any way to avoid this. can you please suggest, if you think of any way.

Reply:
Hi Akshay,

I was thinking we can just invoke the other constructor with a call to `this(<args>)` and have the other constructor do the work. But let's not actually bother bc it looks like the other constructor is meant for testing, and this is only a few lines repeated.